clpy/backend/memory.pyx

```diff
@@ -272,17 +274,46 @@ cdef class MemoryPointer:
             size (int): Size of the sequence in bytes.
 
         """
+        cdef void* tmp
         if size > 0:
-            clpy.backend.opencl.api.EnqueueCopyBuffer(
-                command_queue=clpy.backend.opencl.env.get_command_queue(),
-                src_buffer=src.buf.ptr,
-                dst_buffer=self.buf.ptr,
-                src_offset=src.cl_mem_offset(),
-                dst_offset=self.cl_mem_offset(),
-                cb=size,
-                num_events_in_wait_list=0,
-                event_wait_list=<cl_event*>NULL,
-                event=<cl_event*>NULL)
+            if src.device == self.device:
+                clpy.backend.opencl.api.EnqueueCopyBuffer(
```

Comment:
No need `with self.device` at this line?

Reply:
Oh, it's needed. I modified.